source/extensions/filters/http/ext_proc/on_processing_response.h

```diff
@@ -0,0 +1,56 @@
+#pragma once
+
+#include <memory>
+#include <string>
+
+#include "envoy/config/typed_config.h"
+#include "envoy/server/factory_context.h"
+#include "envoy/service/ext_proc/v3/external_processor.pb.h"
+#include "envoy/stream_info/stream_info.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace ExternalProcessing {
+
+class OnProcessingResponse {
+public:
+  virtual ~OnProcessingResponse() = default;
+  virtual void
+  afterProcessingRequestHeaders(const envoy::service::ext_proc::v3::ProcessingResponse& response,
```

Comment:
This is an interface and should probably have comments describing when the callbacks are invoked.

Also, not sure if it makes sense to have now, but there might be a need for a PassThrough class, that implements all the methods in a naive way. This way, one can override only the needed methods.

Reply:
Added comments.